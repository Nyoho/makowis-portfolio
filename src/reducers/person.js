// @flow
import person from '../models/Person';
import type { Person } from '../models/Person';

const CHANGE_PERSON: 'CHANGE_PERSON' = 'CHANGE_PERSON';

export type State = Person;

type Action = { type: typeof CHANGE_PERSON, person: Person };

const reducer = (state: Person = person, action: Action): State => {
  switch (action.type) {
    case CHANGE_PERSON:
      return action.person;
    default:
      return state;
  }
};

export default reducer;
