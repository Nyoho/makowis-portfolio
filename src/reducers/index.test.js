/* eslint-env jest */
// @flow
import reducer from './index';
import person, { moriC } from '../models/Person';

it('root reducer', () => {
  const state = reducer(
    { person, slides: [] },
    { type: 'CHANGE_PERSON', person },
  );
  expect(state).toMatchSnapshot();
});

it('root reducer change other person', () => {
  const state = reducer(
    { person, slides: [] },
    { type: 'CHANGE_PERSON', person: moriC },
  );
  expect(state).toMatchSnapshot();
});
