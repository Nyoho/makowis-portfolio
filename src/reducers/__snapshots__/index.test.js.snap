// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root reducer 1`] = `
Object {
  "person": Object {
    "avatar": "avatar.bmp",
    "facebook": "makoto.henmi",
    "github": "makowis",
    "id": "mako_wis",
    "name": "Makoto Henmi",
    "nickname": "mako_wis",
    "speakerdeck": "makowis",
    "twitter": "mako_wis",
  },
  "slide": Object {
    "slides": Array [],
  },
}
`;

exports[`root reducer change other person 1`] = `
Object {
  "person": Object {
    "avatar": "https://pbs.twimg.com/profile_images/471280957119406080/LCUoaGga_400x400.png",
    "facebook": "tomohiro.m0219",
    "github": "moriC",
    "id": "moriC",
    "name": "Tomohiro Morishita",
    "nickname": "もりしー",
    "twitter": "CentBoss",
  },
  "slide": Object {
    "slides": Array [],
  },
}
`;
